server/raft.go

```diff
@@ -3474,14 +3474,21 @@ func (n *raft) processAppendEntry(ae *appendEntry, sub *subscription) {
 
 	// Are we receiving from another leader.
 	if n.State() == Leader {
-		// If we are the same we should step down to break the tie.
 		if lterm >= n.term {
 			// If the append entry term is newer than the current term, erase our
 			// vote.
 			if lterm > n.term {
 				n.term = lterm
 				n.vote = noVote
 				n.writeTermVote()
+			} else {
+				assert.Unreachable(
+					"Two leaders using the same term",
+					map[string]any{
+						"term":      n.term,
+						"node id":   n.id,
+						"leader id": ae.leader,
```

Comment:
Could you also include state like `ae.term` and `ae.lterm`?

Reply:
Done